openff/toolkit/typing/engines/smirnoff/forcefield.py

```diff
@@ -1302,8 +1302,12 @@ def create_openmm_system(self, topology, **kwargs):
         if topology.box_vectors is not None:
             system.setDefaultPeriodicBoxVectors(*topology.box_vectors)
 
-        # Add particles (both atoms and virtual sites) with appropriate masses
-        for atom in topology.topology_particles:
+        # Add atom particles with appropriate masses
+        # Virtual site particle creation is handled in the parameter handler
+        # create_force call
+        # This means that even though virtual sites may have been created via
+        # the molecule API, an empty VirtualSites tag must exist in the FF
+        for atom in topology.topology_atoms:
             system.addParticle(atom.atom.mass)
```

Comment:
>  I suppose this was not an issue when creating virtual sites directly from the XML, since the topology only had atoms when the vdW and NonBonded handlers were called (Vsites have them in their dependencies).

Ugggh this took me so many read-throughs to understand, but now it makes perfect sense.